src/container/cli.c

```diff
@@ -1785,45 +1772,29 @@ cont_oid_alloc_complete(tse_task_t *task, void *data)
 
 	if (daos_rpc_retryable_rc(rc) || rc == -DER_STALE) {
 		tse_sched_t *sched = tse_task2sched(task);
-		daos_pool_query_t *pargs;
 		tse_task_t *ptask;
+		unsigned int map_version = out->coao_op.co_map_version;
 
-		/** pool map update task */
-		rc = dc_task_create(dc_pool_query, sched, NULL, &ptask);
+		/** pool map refresh task */
+		rc = dc_pool_create_map_refresh_task(pool, map_version, sched,
+						     &ptask);
 		if (rc != 0)
 			D_GOTO(out, rc);
 
-		pargs = dc_task_get_args(ptask);
-		pargs->poh = arg->coaa_cont->dc_pool_hdl;
-		D_ALLOC_PTR(pargs->info);
-		if (pargs->info == NULL) {
-			dc_task_decref(ptask);
-			D_GOTO(out, rc = -DER_NOMEM);
-		}
-
-		rc = dc_task_reg_comp_cb(ptask, pool_query_cb, NULL, 0);
-		if (rc != 0) {
-			D_FREE(pargs->info);
-			dc_task_decref(ptask);
-			D_GOTO(out, rc);
-		}
-
 		rc = dc_task_resched(task);
 		if (rc != 0) {
-			D_FREE(pargs->info);
-			dc_task_decref(ptask);
+			dc_pool_abandon_map_refresh_task(ptask);
 			D_GOTO(out, rc);
 		}
 
 		rc = dc_task_depend(task, 1, &ptask);
```

Comment:
minor, dc_pool_create_map_refresh_task() creates tse task, maybe can use tse_task_xxx() for L1790 and L1784, although they are same as dc_task_xxx for _resched/_depend

Reply:
Yeah. See my reply to Ken's comment on this above. The dilemma is "task" is a dc_task, while "ptask" is a tse_task. @wangdi1?